libraries/postgres/test/helper.js

```diff
@@ -0,0 +1,40 @@
+const {Client} = require('pg');
+const dbUrl = process.env.TEST_DB_URL;
+
+/**
+ * dbSuite(..) is a replacement for suite(..) that sets this.dbUrl when
+ * a dbUrl exists, or skips when none is available.
+ */
+if (dbUrl) {
+  exports.dbSuite = (...args) => {
+    suite(...args.slice(0, -1), function() {
+      suiteSetup('setup database', function() {
+        exports.dbUrl = dbUrl;
+      });
+      setup('clear database', async function() {
+        await clearDb(dbUrl);
+      });
+      args[args.length - 1].call(this);
```

Comment:
Am I right in thinking that `...args.slice(0, -1)` essentially serves as `args[0]` here?
I'm a little less sure about `args[args.length - 1].call(this)`, but I suppose that runs the suite itself?

Reply:
Not quite, the [negative index is from the end of the array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice) so `slice(0, -1)` gets all but the last element.  And yes, `args[args.length - 1].call(this)` calls that last argument, which is the function defining the suite.